nifi-nar-bundles/nifi-gcp-bundle/nifi-gcp-processors/src/main/java/org/apache/nifi/processors/gcp/bigquery/BigQueryUtils.java

```diff
@@ -37,29 +38,51 @@
     private final static Type gsonSchemaType = new TypeToken<List<Map>>() { }.getType();
 
     public static Field mapToField(Map fMap) {
-        String typeStr = fMap.get("type").toString();
+        String typeStr = fMap.get("type").toString().toUpperCase();
         String nameStr = fMap.get("name").toString();
-        String modeStr = fMap.get("mode").toString();
+        String modeStr;
+        if(fMap.containsKey("mode")) {
+        	modeStr = fMap.get("mode").toString();
+        } else {
+        	modeStr = Mode.NULLABLE.name();
+        }
         LegacySQLTypeName type = null;
+        List<Field> subFields = new ArrayList<>();
 
-        if (typeStr.equals("BOOLEAN")) {
-            type = LegacySQLTypeName.BOOLEAN;
-        } else if (typeStr.equals("STRING")) {
-            type = LegacySQLTypeName.STRING;
-        } else if (typeStr.equals("BYTES")) {
-            type = LegacySQLTypeName.BYTES;
-        } else if (typeStr.equals("INTEGER")) {
-            type = LegacySQLTypeName.INTEGER;
-        } else if (typeStr.equals("FLOAT")) {
-            type = LegacySQLTypeName.FLOAT;
-        } else if (typeStr.equals("TIMESTAMP") || typeStr.equals("DATE")
-                || typeStr.equals("TIME") || typeStr.equals("DATETIME")) {
-            type = LegacySQLTypeName.TIMESTAMP;
-        } else if (typeStr.equals("RECORD")) {
-            type = LegacySQLTypeName.RECORD;
+        switch(typeStr) {
+        case "BOOLEAN":
+        	type = LegacySQLTypeName.BOOLEAN;
+        	break;
+        case "STRING":
+        	type = LegacySQLTypeName.STRING;
+        	break;
+        case "BYTES":
+        	type = LegacySQLTypeName.BYTES;
+        	break;
+        case "INTEGER":
+        	type = LegacySQLTypeName.INTEGER;
+        	break;
+        case "FLOAT":
+        	type = LegacySQLTypeName.FLOAT;
+        	break;
+        case "RECORD":
+            List<Map> fields = (List<Map>) fMap.get("fields");
+        	type = LegacySQLTypeName.RECORD;
```

Comment:
Could you please move this assignment statement to the top of this case block? The code logic would be more straightforward in that way.